### ISSUE TITLE
Does the default EKS AMI provide IO stats for cgroups for cadvisor?

### ISSUE THREAD

**Issue by badaldavda8** [awslabs/amazon-eks-ami#217]
**Environment**:
- AWS Region:
- Instance Type(s): 
- EKS Platform version (use `aws eks describe-cluster --name <name> --query cluster.platformVersion`): "eks.2"
- Kubernetes version (use `aws eks describe-cluster --name <name> --query cluster.version`): 1.11
- AMI Version: latest ami
- Kernel (e.g. `uname -a`): 
- Release information (run `cat /tmp/release` on a node):

```
```

Trying to setup prometheus with cadvisor and wants to understand if we allow default EKS AMI provide IO stats for cgroups? 

To verify this, I tried to do docker stats since cadvisor takes metrics from docker stats - I get the following output - 
a507a6073aea 0.02% 88.4MiB / 3.796GiB 2.27% 0B / 0B 0B / 13.2GB 12 
f37649a59495 0.00% 11.12MiB / 3.796GiB 0.29% 0B / 0B 0B / 0B 8 
5776d73d9284 0.00% 2.672MiB / 3.796GiB 0.07% 0B / 0B 0B / 0B 2 
230300ce2da3 0.00% 304KiB / 3.796GiB 0.01% 0B / 0B 0B / 0B 1 
0baaa8c8a78a 0.00% 368KiB / 3.796GiB 0.01% 0B / 0B 57.3kB / 0B 1 
321517ff92ec 0.00% 288KiB / 3.796GiB 0.01% 0B / 0B 0B / 0B 1 
b306328d2b5e 0.00% 404KiB / 3.796GiB 0.01% 0B / 0B 0B / 0B 1 
1354bfbfb46c 0.00% 204KiB / 3.796GiB 0.01% 0B / 0B 0B / 0B 1 
5e2f50f9e314 0.10% 12.28MiB / 3.796GiB 0.32% 0B / 0B 0B / 84.4MB 10 

whereas this is the actual output that should be seen - 
https://docs.docker.com/engine/reference/commandline/stats/ 

Does not even work for network along with io. 

Trying to get this "container_fs_write_seconds_total" from cadivsor. 

Tried replicating but prometheus is giving error for kubelet saying timeout.

Basically the query is why does docker stat not show network and io stats? Does the default EKS AMI provide IO stats for cgroups?

**Comment by micahhausler** [comment]
I'm also seeing all zeros for `container_fs_write_seconds_total` when I scrape a kubelet's `/metrics/cadvisor` endpoint. Here's the Prometheus config I used

```yaml
scrape_configs:
- job_name: kubernetes-nodes-cadvisor
  scrape_interval: 1m
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: https
  kubernetes_sd_configs:
  - role: node
  bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
  tls_config:
    ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    insecure_skip_verify: true
  relabel_configs:
  - separator: ;
    regex: __meta_kubernetes_node_label_(.+)
    replacement: $1
    action: labelmap
  - separator: ;
    regex: (.*)
    target_label: __address__
    replacement: kubernetes.default.svc:443
    action: replace
  - source_labels: [__meta_kubernetes_node_name]
    separator: ;
    regex: (.+)
    target_label: __metrics_path__
    replacement: /api/v1/nodes/$1/proxy/metrics/cadvisor
    action: replace
```

This looks like an upstream cadvisor issue, that would be the best place to track this issue.

kubernetes/kubernetes#73000
google/cadvisor#2174
google/cadvisor#2148

Since this isn't an EKS-specific AMI issue, I'm going to close this out
